### PR TITLE
Adds support for swipe actions

### PR DESCRIPTION
**Pull request by simonmitchell**
Adds support for swipe actions via `RowActionable`,`SwipeActionsConfigurable` and `Row/Section` protocols

### CODE REVIEW

File: ThunderTable/TableRow+Actions.swift [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		case .default:
+			return .default
+		case .normal:
+			return .normal
+		}
+	}
+}
+
+/// A closure called when a `RowActionable` is triggered by the user
+/// call the completionHandler (iOS 11 and above) to reset the context to its normal state (e.g. when swiping, resets to unswiped state)
```
Comment (ryanbourneuk) [inline ThunderTable/TableRow+Actions.swift]: If this is a separate sentence to the one above, could it start with a capital please? Got a bit confused as I thought it was one sentence to begin with :)

File: ThunderTable/TableSection.swift [hunk 2]
```diff
@@ -23,6 +23,10 @@ public protocol Section {
 	var editHandler: EditHandler? { get set }
     
     var selectionHandler: SelectionHandler? { get set }
+	
+	var rowLeadingSwipeActionsConfiguration: SwipeActionsConfigurable? { get }
```
Comment (ryanbourneuk) [inline ThunderTable/TableSection.swift]: Indentation seems a bit off here, guessing Github has got confused with tabs and spaces again? :)

**Review by ryanbourneuk — APPROVED** [review]

File: ThunderTable/TableRow+Actions.swift [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	}
+	
+	/// Creates a `UITableViewRowAction` to be used in the delegate method of `UITableViewController` with the provided handler
+	///
+	/// - Parameter handler: The handler to be called
+	/// - Returns: A `UIContextualAction`
+	func rowAction(with handler: @escaping (UITableViewRowAction, IndexPath) -> Void) -> UITableViewRowAction {
+		let rowAction = UITableViewRowAction(style: style._UITableViewRowActionStyle, title: title, handler: handler)
+		rowAction.backgroundColor = backgroundColor
+		rowAction.backgroundEffect =  backgroundEffect
```
Comment (MattCheetham) [inline ThunderTable/TableRow+Actions.swift]: Double space after the `=` ?

**Review by MattCheetham — CHANGES_REQUESTED** [review]

**Comment by simonmitchell** [comment]
@MattCheetham @ryanbourneuk changes made

**Review by MattCheetham — APPROVED** [review]

**Pull request merged**